models/facet.js

```diff
@@ -132,7 +134,15 @@ Facet.loadIntoCache = function(cache, facetName, allConfigFiles, cb) {
         debug('loading [%s] dataSource into cache', dataSourceName);
         DataSourceDefinition.addToCache(cache, def);
       });
-      
+      cb();
+    });
+  }
+
+  if(middlewares) {
+    steps.push(function(cb) {
+      middlewares.load(cb);
+    }, function(cb) {
+      Middleware.deserialize(cache, facetName, middlewares);
```

Comment:
Other entities have the logic embedded in facet.js, not as separate functions on the entity model itself.